### PR TITLE
docs: Disambiguate "OpenShift", rename files and IDs

### PR DESCRIPTION
**Pull request by robin-owen**
## Solution/Idea

Use the `{ocp}` and `{openshift}` attributes in the documentation where relevant. This ensures that we disambiguate between the product itself (CRC, downstream OpenShift Local) and the OpenShift Container Platform cluster managed by it. At the same time, we avoid the potential for unfortunate typos by using attributes everywhere in the documentation.

As an additional task that came up while renaming, this PR also contains a commit to rename the source files and IDs of the documentation to avoid direct references to the name of the product. This is largely so that we can use the exact same source files across both upstream and downstream documentation (which have two very, very different names).

The changes contained in this PR appear large, but mostly consist of renaming file names and section IDs. Other changes involve replacing the literal text of "OpenShift" with either "`{ocp}`" or "`{openshift}`" (depending on context).

### CODE REVIEW

File: docs/source/topics/ref_dns-configuration.adoc [hunk 1]
```diff
@@ -4,11 +4,11 @@
...
 
-The OpenShift cluster managed by {prod} uses 2 DNS domain names, `crc.testing` and `apps-crc.testing`.
-The `crc.testing` domain is for core OpenShift services.
-The `apps-crc.testing` domain is for accessing OpenShift applications deployed on the cluster.
+The {ocp} cluster managed by {prod} uses 2 DNS domain names, `crc.testing` and `apps-crc.testing`.
+The `crc.testing` domain is for core {ocp} services.
+The `apps-crc.testing` domain is for accessing {openshift} applications deployed on the cluster.
 
-For example, the OpenShift API server is exposed as `api.crc.testing` while the OpenShift console is accessed as `console-openshift-console.apps-crc.testing`.
+For example, the {ocp} API server is exposed as `api.crc.testing` while the {ocp} console is accessed as `console-openshift-console.apps-crc.testing`.
```
Comment (robin-owen) [inline docs/source/topics/ref_dns-configuration.adoc:11]: @code-ready/crc-team  Is it accurate to refer to the OpenShift Container Platform API server, or should we leave this as OpenShift API server?

File: docs/source/topics/assembly_accessing-the-openshift-cluster.adoc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [id="accessing-the-openshift-cluster_{context}"]
-= Accessing the OpenShift cluster
+= Accessing the {openshift} cluster
```
Comment (cfergeau) [inline docs/source/topics/assembly_accessing-the-openshift-cluster.adoc:2]: Just below, it is `Access the {ocp} cluster`, and here it is an `{openshift} cluster`

File: docs/source/topics/assembly_troubleshooting.adoc [hunk 3]
```diff
@@ -3,10 +3,9 @@
 
 [NOTE]
 ====
-The goal of {rh-prod} is to deliver an OpenShift environment for development and testing purposes.
-Issues occurring during installation or usage of specific OpenShift applications are outside of the scope of {prod}.
+The goal of {rh-prod} is to deliver an {ocp} environment for development and testing purposes.
+Issues occurring during installation or usage of specific {openshift} applications are outside of the scope of {prod}.
 Report such issues to the relevant project.
-For example, OpenShift tracks issues on link:https://github.com/openshift/origin/issues[GitHub].
```
Comment (cfergeau) [inline docs/source/topics/assembly_troubleshooting.adoc:9]: Removal of this line is intentional?

File: docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc [hunk 4]
```diff
@@ -1,18 +1,18 @@
 [id="accessing-the-internal-openshift-registry_{context}"]
-= Accessing the internal OpenShift registry
+= Accessing the internal {openshift} registry
 
-The OpenShift cluster running in the {prod} instance includes an internal container image registry by default.
+The {ocp} cluster running in the {prod} instance includes an internal container image registry by default.
 This internal container image registry can be used as a publication target for locally developed container images.
-To access the internal OpenShift registry, follow these steps.
+To access the internal {ocp} registry, follow these steps.
```
Comment (cfergeau) [inline docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc:6]: Was `{openshift} registry` in the title, and here `{ocp} registry`

File: docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 .Prerequisites
 
 * {prod} is configured to use the {openshift} preset.
 For more information, see link:{crc-gsg-url}#changing-the-selected-preset_gsg[Changing the selected preset].
 * A running {prod} instance.
 For more information, see link:{crc-gsg-url}#starting-the-instance_gsg[Starting the instance].
-* A working OpenShift CLI ([command]`oc`) command.
-For more information, see link:{crc-gsg-url}#accessing-the-openshift-cluster-with-oc_gsg[Accessing the OpenShift cluster with the OpenShift CLI].
+* A working {openshift} CLI ([command]`oc`) command.
+For more information, see link:{crc-gsg-url}#accessing-the-openshift-cluster-with-oc_gsg[Accessing the {openshift} cluster with the {openshift} CLI].
```
Comment (cfergeau) [inline docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc:15]: was `{ocp} cluster` before in the PR

File: docs/source/topics/proc_accessing-the-openshift-cluster-with-oc.adoc [hunk 6]
```diff
@@ -50,8 +50,8 @@ $ oc get co
 {prod} disables the Cluster Monitoring Operator by default.
 ====
 
-See link:{crc-gsg-url}#troubleshooting-codeready-containers_gsg[Troubleshooting {prod}] if you cannot access the {prod} OpenShift cluster.
+See link:{crc-gsg-url}#troubleshooting-codeready-containers_gsg[Troubleshooting {prod}] if you cannot access the {prod} cluster.
```
Comment (cfergeau) [inline docs/source/topics/proc_accessing-the-openshift-cluster-with-oc.adoc]: Just `{prod}`, no `{ocp}` or `{openshift}`?

File: docs/source/topics/proc_accessing-the-openshift-web-console.adoc [hunk 7]
```diff
@@ -31,8 +31,8 @@ You can also view the password for the `developer` and `kubeadmin` users by runn
 $ {bin} console --credentials
 ----
 
-See link:{crc-gsg-url}#troubleshooting-codeready-containers_gsg[Troubleshooting {prod}] if you cannot access the {prod} OpenShift cluster.
+See link:{crc-gsg-url}#troubleshooting-codeready-containers_gsg[Troubleshooting {prod}] if you cannot access the {prod} cluster.
```
Comment (cfergeau) [inline docs/source/topics/proc_accessing-the-openshift-web-console.adoc]: ditto, just `{prod}`?

File: docs/source/topics/proc_connecting-to-remote-instance.adoc [hunk 8]
```diff
@@ -63,4 +63,4 @@ $ sudo systemctl reload NetworkManager
 $ oc login -u developer -p developer https://api.crc.testing:6443
 ----
 +
-The remote OpenShift Web Console is available at \https://console-openshift-console.apps-crc.testing.
+The remote {ocp} web console is available at \https://console-openshift-console.apps-crc.testing.
```
Comment (cfergeau) [inline docs/source/topics/proc_connecting-to-remote-instance.adoc:66]: so far it was `{openshift} web console`

File: docs/source/topics/ref_dns-configuration.adoc (hunk 1 above)
Comment (cfergeau) [reply docs/source/topics/ref_dns-configuration.adoc:11]: I have a hard time differentiating between the 2 names. OCP API server seems fine, but I would not mind the other one eithre ;)

File: docs/source/topics/assembly_troubleshooting.adoc [hunk 9]
```diff
@@ -3,10 +3,9 @@
 
 [NOTE]
 ====
-The goal of {rh-prod} is to deliver an OpenShift environment for development and testing purposes.
-Issues occurring during installation or usage of specific OpenShift applications are outside of the scope of {prod}.
+The goal of {rh-prod} is to deliver an {ocp} environment for development and testing purposes.
```
Comment (cfergeau) [inline docs/source/topics/assembly_troubleshooting.adoc:6]: `{rh-prod}` currently expands to `Red Hat CRC` which is not what we want.

Comment (robin-owen) [reply docs/source/topics/assembly_troubleshooting.adoc:6]: Upstream, correct. Downstream, we have both `{prod}` and `{rh-prod}` defined as the exact same content: "Red Hat OpenShift Local." There's no duplication of the term across versions.

Are you suggesting that we shouldn't mention "Red Hat CRC" at all? If so, we can redefine the contents of the attribute in `meta/product-attributes.adoc`.

File: docs/source/topics/proc_connecting-to-remote-instance.adoc (hunk 8 above)
Comment (robin-owen) [reply docs/source/topics/proc_connecting-to-remote-instance.adoc:66]: This change was based on how they write about the web console in the docs for OCP. There, they always mention OCP explicitly, and I assume that's the best practice for us as well.

File: docs/source/topics/proc_accessing-the-openshift-cluster-with-oc.adoc (hunk 6 above)
Comment (robin-owen) [reply docs/source/topics/proc_accessing-the-openshift-cluster-with-oc.adoc]: Good point. Perhaps we should say "The {ocp} cluster managed by {prod}".

File: docs/source/topics/assembly_troubleshooting.adoc (hunk 3 above)
Comment (robin-owen) [reply docs/source/topics/assembly_troubleshooting.adoc:9]: It is, mostly due to the fact that I honestly don't know where we should direct users for issues upstream. The repository we originally linked to seems to have a completely different focus now. Should we link to OKD for upstream issue reporting?

File: docs/source/topics/assembly_accessing-the-openshift-cluster.adoc (hunk 2 above)
Comment (robin-owen) [reply docs/source/topics/assembly_accessing-the-openshift-cluster.adoc:2]: Correct. I tried to keep the titles short by only referring to OpenShift, but using the complete name (OpenShift Container Platform) in the text itself.

File: docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc (hunk 5 above)
Comment (robin-owen) [reply docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc:15]: This is due to the titling. We're using `{openshift}` in titles to keep them short, `{ocp}` in the body text of the docs for specificity. The link is titled after the name of the section, which uses the shortened version.

File: docs/source/topics/proc_accessing-the-openshift-web-console.adoc (hunk 7 above)
Comment (robin-owen) [reply docs/source/topics/proc_accessing-the-openshift-web-console.adoc]: Changing to "The {ocp} cluster managed by {prod}".

File: docs/source/topics/ref_dns-configuration.adoc (hunk 1 above)
Comment (gbraad) [reply docs/source/topics/ref_dns-configuration.adoc:11]: In this case it is not a necessity.

File: docs/source/topics/assembly_troubleshooting.adoc (hunk 9 above)
Comment (gbraad) [reply docs/source/topics/assembly_troubleshooting.adoc:6]: upstream is just `CRC`

**Review by gbraad — APPROVED** [review]

File: docs/source/topics/assembly_troubleshooting.adoc (hunk 9 above)
Comment (robin-owen) [reply docs/source/topics/assembly_troubleshooting.adoc:6]: Updated this PR to remove the "Red Hat" from "Red Hat CRC".

**Comment by robin-owen** [comment]
@cfergeau @gbraad Is there anything more that needs to be addressed before we can merge this?

**Comment by gbraad** [comment]
I approved. Since no further comments i will merge now

**Pull request merged**